### PR TITLE
A working instruction reader. Processes of binary instructions into Instruction objects

### PR DESCRIPTION
**Pull request by toon-leader-bacon**

### CODE REVIEW

File: src/main/java/group1/mips_simulator/Utility.java
Comment (akm303) [inline src/main/java/group1/mips_simulator/Utility.java:1]: minor thing that wont affect anything
consider renaming "octal" variable to "octalString" or something, just to be explicit about what value is being passed into some of these functions

File: src/main/java/group1/mips_simulator/components/instruction/Field.java [hunk 1]
```diff
@@ -1,4 +1,91 @@
...
+            return false;
+        }
+
+        final Field other = (Field) obj;
+        return this.value == other.value &&
+                this.size == other.size;
+    }
+
+    @Override
+    public int hashCode() {
```
Comment (akm303) [inline src/main/java/group1/mips_simulator/components/instruction/Field.java:85]: what is the purpose of hash?

File: src/main/java/group1/mips_simulator/Utility.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        return result.toString();
+    }
+
+    /**
+     * Convert the octal string number into a binary number representation of the
+     * target size. 0s will be padded to hit the target, or only the 'targetSize'
+     * least significant bits will be returned.
+     * IE: The most significant bits will cut in that case.
+     */
+    public static String octalToBinary(String octal, int targetSize) {
```
Comment (akm303) [inline src/main/java/group1/mips_simulator/Utility.java:40]: minor thing that wont affect anything
consider renaming "octal" variable to "octalString" or something, just to be explicit about what value is being passed into some of these functions

**Review by akm303 — APPROVED** [review]
looks good, left some minor comments/questions for when you get a chance, but shouldnt affect its operation

File: src/main/java/group1/mips_simulator/components/instruction/Field.java (hunk 1 above)
Comment (toon-leader-bacon) [reply src/main/java/group1/mips_simulator/components/instruction/Field.java:85]: Any time you override the `.equals` function you really _really_ should override the `.hashCode()` function too to keep the invariance that: "Two objects are `.equal(...)` if and only if they have the same `.hashCode()`. 

There are some built in classes that expect this behavior, like the `HashMap<...>`

File: src/main/java/group1/mips_simulator/Utility.java (hunk 2 above)
Comment (toon-leader-bacon) [reply src/main/java/group1/mips_simulator/Utility.java:40]: Ok! I consider the types of the params to be part of the function signature so adding the word "String" seems redundant to me. But that's just a minor style thing so I'll add it now :)

Now if we were in a weakly typed language like python I'd 100% agree with you.

Comment (akm303) [reply src/main/java/group1/mips_simulator/Utility.java]: thats fair, thats fair

**Pull request merged**